### ISSUE TITLE
Custom Audio files not working

### ISSUE THREAD

**Issue by cefranco76** [TECH7Fox/asterisk-hass-addons#245]
Hi @felipecrs 
I'm trying to use pt-BR in my addon but it's not working.
I've downloaded pt-BR files as mentioned in documentation and at folder 'media/asterisk/sounds/pt_br'

I've set CHANNEL language that way:
Set(CHANNEL(language)=pt_br)

I play the audio in extension:
Playback(welcome)

We can see in LOG:
<PJSIP/6001-0000004c> Playing 'welcome.alaw' (language 'pt_br')

But it's in english.

Would you please help me to understand if is there an issue no addon or my code? Thank you very much!

Good job on this addon, you guys!!

**Comment by felipecrs** [comment]
Hi @cefranco76. To be completely honest with you I did not test this feature. I actually don't even know how to use it either.

- It may be completely broken
- Or maybe we just don't know how to use

I will test it when I find some time, but just so you know, I use the Google Translate:

https://zaf.github.io/asterisk-googletts/

It's installed by default in the add-on.

**Comment by cefranco76** [comment]
> Hi @cefranco76. To be completely honest with you I did not test this feature. I actually don't even know how to use it either.
> 
> * It may be completely broken
> * Or maybe we just don't know how to use
> 
> I will test it when I find some time, but just so you know, I use the Google Translate:
> 
> https://zaf.github.io/asterisk-googletts/
> 
> It's installed by default in the add-on.

Sure, I'll try, thanks, but I don't know how to paste a file in that folder, or even if I've got access to it through addon.
 /var/lib/asterisk/agi-bin/
 
 Did you know if is there a Edgetts AGI script? Edge TTS has a more natural voice.
 
 Thanks again!! Valeu amigo, sou brazuca também.

**Comment by cefranco76** [comment]
> As I said, my friend, this AGI script is already installed in the add-on. You can simply call it from your extensions.

It worked. You can close the issue, if you like, thanks!

**Comment by felipecrs** [comment]
Awesome!

But I'll leave it open... I want to make sure the audio files works.

**Comment by sesardelaisla** [comment]
Just for reference, my audios also play in English, despite I have es_ES references in my config.

Apart from that, may I ask whether it is possible to call such AGI script from an `originate` CLI command? I would like to be able to play a message in Spanish upon placing a call. Being able to change the message from a service stdin call would be great because it would be very easy to customize without needing to reload the dialplan everytime you wish to change the message. If something like this is possible (but changing the tt-monkeys Asterisk sample sound by the Spanish sentence plus the country string) would be great:

```
service: hassio.addon_stdin
data:
  addon: 3e533915_asterisk
  input: originate Local/[phone_number]@external-calls application Playback tt-monkeys
```

**Comment by felipecrs** [comment]
> Just for reference, my audios also play in English, despite I have es_ES references in my config.

By the way, PRs are welcome to fix this issue.

**Comment by felipecrs** [comment]
Ok, I did some testing, and this is working for me.

I have `pt-BR` set in the add-on configs to download. I can confirm the files were downloaded by checking the `/media/asterisk/sounds/pt_BR` folder.

I tested with `chan-dongle`:

```ini
; /config/asterisk/custom/dongle.conf
[defaults]
language=pt_BR
```

With some things like:

```ini
; /config/asterisk/custom/extensions.conf
[dongle-incoming]
; Coming from dongle
exten => <dongle-number>,1,Set(GLOBAL(PHONE_RETURN_TO_BRIDGE)=0)
 same => n,GoTo(s,1)

exten => s,1,Answer()

 same => n,Wait(5)
 same => n,Playback(you-wish-to-join)
 same => n,Playback(wx/percent)

 same => n,Wait(5)
 same => n,Hangup()
```

@sesardelaisla I even tested with `es_ES` and it's working as well.

Something I noticed though is that `Set(LANGUAGE()=pt_BR)` does not seem to work for some reason. Only by changing the dongle config. But I assume this problem only affects chan dongle.

I'm making PR #271 which should be a good thing, but be aware that my tests are working even without it.

In other words, I can't reproduce this issue. Nevertheless, if you can reproduce this issue, would you mind checking if any of the following solutions make your `es_ES` audio files work?

## First solution

```console
docker exec -it addon_3e533915_asterisk bash -c 'rm -rf /var/lib/asterisk/sounds/es_ES && cp -arf /media/asterisk/sounds/es_ES /var/lib/asterisk/sounds/es_ES'
```

## Second solution

This one is the same as #271.

```console
docker exec -it addon_3e533915_asterisk bash -c 'chmod -R 0755 /var/lib/asterisk/sounds/es_ES'
```

Unfortunately I'm not sure how to test this otherwise.

**Comment by felipecrs** [comment]
I found something weird though in my logs:

```
ERROR[6385][C-00000008]: pbx_functions.c:700 ast_func_write: Function LANGUAGE not registered
```

**Comment by felipecrs** [comment]
Ok, this is how it should be done in the extensions.conf:

```
same => n,Set(CHANNEL(language)=es_ES)
```

**Comment by felipecrs** [comment]
@cefranco76 I tried your case as well (welcome file) and it's working properly.

I will release a new version of the add-on later today. Please remove all your sound files:

```
rm -rf /media/asterisk/sounds/pt_*
```

Then add `pt-BR` to the add-on configuration, restart it, wait for it to download and convert all pt-BR files and you can use like following:

```
exten => s,1,Answer()

 same => n,Set(CHANNEL(language)=pt_BR)

 same => n,Wait(5)
 same => n,Playback(welcome)

 same => n,Wait(5)
 same => n,Hangup()
```

**Comment by felipecrs** [comment]
I will close this issue for now, unless some of you provide me more information to work with. :)

**Comment by sesardelaisla** [comment]
> would you mind checking if any of the following solutions make your `es_ES` audio files work?

Happy to test whatever it is necessary to fix this! My previous test was with the following string in my `pjsip.conf` file only:

```
[general]
language=es_ES
```

Now, I changed `dongle.conf` value `language=en` to `language=es_ES`. Still getting English sounds after restart. Is there anything I should do for the purpose of the test? Not sure whether I should do something in `extensions.conf`.

**Comment by sesardelaisla** [comment]
> same => n,Set(CHANNEL(language)=es_ES)

Sorry. Missed this. I included such string, re-downloaded the es_ES sounds and everything is working all right now. Thanks!

Apart from this, for consistency reasons, I suggest to change the way the string should be included in the config from es-ES to es_ES.

**Comment by felipecrs** [comment]
I agree with the config change. The trick part is the automatic migration for existing users.

**Comment by cefranco76** [comment]
> rm -rf /media/asterisk/sounds/pt_*



> @cefranco76 I tried your case as well (welcome file) and it's working properly.
> 
> I will release a new version of the add-on later today. Please remove all your sound files:
> 
> ```
> rm -rf /media/asterisk/sounds/pt_*
> ```
> 
> Then add `pt-BR` to the add-on configuration, restart it, wait for it to download and convert all pt-BR files and you can use like following:
> 
> ```
> exten => s,1,Answer()
> 
>  same => n,Set(CHANNEL(language)=pt_BR)
> 
>  same => n,Wait(5)
>  same => n,Playback(welcome)
> 
>  same => n,Wait(5)
>  same => n,Hangup()
> ```

It work properly, thanks, although there're stil some missing files

WARNING[815][C-00000004]: file.c:824 ast_openstream_full: File digits/hours does not exist in any format
WARNING[815][C-00000004]: file.c:1303 ast_streamfile: Unable to open digits/hours (format (ulaw)): No such file or directory

**Comment by felipecrs** [comment]
Can you confirm the files exist in https://www.asterisksounds.org/ for your language? If they don't, there's nothing we can do about it.
